### PR TITLE
Transfer wallet to wallet

### PR DESCRIPTION
**Pull request by simeonpp**

### CODE REVIEW

**Comment by simeonpp** [comment]
removeRedundantInformation method is used in the to remove  information that is specific for a port, e.g.:

ledgerPort might add transfer.message and transfer.type (used for logging purposes in event push) but this information is not relevant for the rest of the ports.

File: api/script/index.js [hunk 1]
```diff
@@ -138,7 +136,7 @@ module.exports = {
             }) // .this is intentionally after catch as we do not want to this.log the original error
             .then(x => Promise.reject(error));
         };
-        var dbPushExecute = transfer => bus.importMethod('db/transfer.push.execute')(transfer)
+        var dbPushExecute = transfer => this.us.importMethod('db/transfer.push.execute')(transfer)
```
Comment (kalinkrustev) [inline api/script/index.js]: change to this.bus

File: api/script/index.js [hunk 2]
```diff
@@ -154,31 +158,35 @@ module.exports = {
...
                     .then(result => {
                         if (transfer.transferType === 'ministatement') {
                             transfer.ministatement = result.ministatement;
                         }
                         transfer.balance = result.balance;
                         transfer.transferIdIssuer = result.transferIdIssuer;
-                        return transfer;
+
+                        result.transferId = transfer.transferId;
+                        result.transferIdIssuer = transfer.transferIdIssuer;
```
Comment (kalinkrustev) [inline api/script/index.js]: redundand code - see 2 lines above

File: api/script/index.js [hunk 3]
```diff
@@ -154,31 +158,35 @@ module.exports = {
...
                 return this.bus.importMethod('db/transfer.push.requestLedger')(transfer)
                     .then(() => transfer)
-                    .then(this.bus.importMethod(transfer.ledgerPort + '/transfer.push.execute'))
+                    .then(this.bus.importMethod(transfer.ledgerPort + '.push.execute'))
                     .catch(handleError(transfer, 'Ledger'))
                     .then(result => {
                         transfer.transferIdLedger = result.transferIdIssuer;
-                        return transfer;
+                        result.transferId = transfer.transferId;
+                        result.transferIdLedger = transfer.transferIdIssuer;
```
Comment (kalinkrustev) [inline api/script/index.js]: redundand code - see 2 lines above

File: api/sql/schema/360$transfer.split.sql [hunk 4]
```diff
@@ -1,13 +1,24 @@
...
+    credit VARCHAR(50) NOT NULL, -- the account number that will be credited
+    amount MONEY, -- the amount of the split
+    [description] VARCHAR(50) NOT NULL, -- split description
+    tag VARCHAR(MAX), -- tags used to identify what the split record refers to (eg. fee, commission)
+    [debitActorId] [bigint] NULL, -- id of the actor whose account will be debited
+    [creditActorId] [bigint] NULL, -- id of the account whose account will be credited
+    [debitItemId] [bigint] NULL, -- id of the item whose account will be credited
+    [creditItemId] [bigint] NULL, -- id of the item whose account will be credited
+    [state] [smallint] NULL, -- status of the split (eg. rejected, paid etc.)
+    [txtId] [bigint] NULL, -- id of the transaction that paid the the split amount
```
Comment (kalinkrustev) [inline api/sql/schema/360$transfer.split.sql]: Can we put a better name, like transferIdPayment ?

File: api/sql/schema/360$transfer.split.sql [hunk 5]
```diff
@@ -1,13 +1,24 @@
...
     CONSTRAINT [pkTransferSplit] PRIMARY KEY CLUSTERED ([splitId] ASC),
-    CONSTRAINT [fkTransferSplit_TransferId] FOREIGN KEY([transferId]) REFERENCES [transfer].[transfer] ([transferId])
+    CONSTRAINT [fkTransferSplit_TransferId] FOREIGN KEY ([transferId]) REFERENCES [transfer].[transfer] ([transferId]),
+    CONSTRAINT [fkTransferSplit_TxtId] FOREIGN KEY ([txtId]) REFERENCES [transfer].[transfer] ([transferId]),
```
Comment (kalinkrustev) [inline api/sql/schema/360$transfer.split.sql]: After renaming field TxtId, fix name of constraint too.

File: api/sql/schema/380$transfer.alterTable.sql [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+IF NOT EXISTS( SELECT 1 FROM sys.columns WHERE Name = N'txtId' AND Object_ID = Object_ID(N'transfer.split') )
+BEGIN
+    ALTER TABLE [transfer].[split] ADD txtId BIGINT
+END
+
+---transfer.split/FKs
+IF NOT EXISTS (SELECT 1 FROM sys.foreign_keys WHERE name = 'fkTransferSplit_TxtId')
+BEGIN
+    ALTER TABLE [transfer].[split] ADD CONSTRAINT fkTransferSplit_TxtId FOREIGN KEY ([txtId]) REFERENCES [transfer].[transfer] ([transferId])
```
Comment (kalinkrustev) [inline api/sql/schema/380$transfer.alterTable.sql]: After renaming field TxtId, fix name of constraint too.

File: api/sql/schema/380$transfer.alterTable.sql [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+BEGIN
+    ALTER TABLE [transfer].[split] ADD creditItemId BIGINT
+END
+
+IF NOT EXISTS( SELECT 1 FROM sys.columns WHERE Name = N'state' AND Object_ID = Object_ID(N'transfer.split') )
+BEGIN
+    ALTER TABLE [transfer].[split] ADD [state] SMALLINT
+END
+
+IF NOT EXISTS( SELECT 1 FROM sys.columns WHERE Name = N'txtId' AND Object_ID = Object_ID(N'transfer.split') )
```
Comment (kalinkrustev) [inline api/sql/schema/380$transfer.alterTable.sql]: Can we put a better name, like transferIdPayment ?

File: api/sql/schema/750$transfer.push.confirmIssuer.sql [hunk 8]
```diff
@@ -13,4 +16,18 @@ WHERE
     transferId = @transferId AND
     issuerTxState = 1
 
-IF @@ROWCOUNT <> 1 RAISERROR('transfer.confirmIssuer', 16, 1);
+DECLARE @COUNT int = @@ROWCOUNT
+IF @COUNT <> 1 RAISERROR('transfer.confirmIssuer', 16, 1);
```
Comment (kalinkrustev) [inline api/sql/schema/750$transfer.push.confirmIssuer.sql]: raise should be last statement, as we do not want to lose the event
see pattern in other similar procedures

File: api/sql/schema/750$transfer.push.confirmLedger.sql [hunk 9]
```diff
@@ -14,3 +17,16 @@ WHERE
     ledgerTxState = 1
 
 IF @@ROWCOUNT <> 1 RAISERROR('transfer.confirmLedger', 16, 1);
+
```
Comment (kalinkrustev) [inline api/sql/schema/750$transfer.push.confirmLedger.sql]: raise should be last statement, as we do not want to lose the event
see pattern in other similar procedures

File: api/sql/schema/750$transfer.push.confirmMerchant.sql [hunk 10]
```diff
@@ -14,3 +17,17 @@ WHERE
     merchantTxState = 1
 
 IF @@ROWCOUNT <> 1 RAISERROR('transfer.confirmMerchant', 16, 1);
+
```
Comment (kalinkrustev) [inline api/sql/schema/750$transfer.push.confirmMerchant.sql]: raise should be last statement, as we do not want to lose the event
see pattern in other similar procedures

File: api/sql/schema/750$transfer.push.create.sql [hunk 11]
```diff
@@ -180,7 +180,9 @@ BEGIN TRY
             conditionId,
             splitNameId,
             [description],
-            tag
+            tag,
+            [state],
+            creditActorId
```
Comment (kalinkrustev) [inline api/sql/schema/750$transfer.push.create.sql]: please add all the new fields in the result

File: api/sql/schema/750$transfer.transferData.get.sql [hunk 12]
```diff
@@ -0,0 +1,37 @@
+ALTER PROCEDURE [transfer].[transferData.get]
```
Comment (kalinkrustev) [inline api/sql/schema/750$transfer.transferData.get.sql]: Can we rename to transfer.transfer.get?

**Pull request merged**